saltproc/saltproc.py

```diff
@@ -55,103 +71,280 @@ def __init__(self, steps, cores, nodes, bw, exec_path, restart=False,
+        self.driver_before_db[0,
+                              :] = self.dep_dict[self.driver_mat_name] * self.driver_vol
+        self.driver_after_db[0,
+                             :] = self.dep_dict[self.driver_mat_name] * self.driver_vol
+        try:
+            self.blanket_before_db[0,
+                                   :] = self.dep_dict[self.blanket_mat_name] * self.blanket_vol
+            self.blanket_after_db[0,
+                                  :] = self.dep_dict[self.blanket_mat_name] * self.blanket_vol
+        except:
+            self.blanket_before_db[0, :] = np.zeros(self.number_of_isotopes)
+            self.blanket_after_db[0, :] = np.zeros(self.number_of_isotopes)
+            print('Blanket not defined: going to be all zeros')
+
+    def write_run_info(self):
+        """ Reads from the input file to write to hdf5
+            of important SERPENT and Saltproc run parameters
         """
-        array = np.array([])
-        for key, value in bumat_dict.items():
-            array = np.append(array, value)
-        return array
+        # read from input file:
+        with open(self.input_file, 'r') as f:
+            lines = f.readlines()
+            for linenum, line in enumerate(lines):
+                if (line.split('%')[0]).strip() == 'dep':
+                    timestep = (lines[linenum+2].split('%')[0]).strip()
+                    if ' ' in timestep:
+                        raise ValueError(
+                            'Your Input file should only have one depstep')
+                if 'set pop' in line and '%' not in line:
+                    neutrons = int(line.split()[2])
+                    active = int(line.split()[3])
+                    inactive = int(line.split()[4])
+        # write to db
+        self.f.create_dataset('siminfo_timestep', data=timestep)
+        self.f.create_dataset('siminfo_pop', data=[neutrons, active, inactive])
+
+        self.f.create_dataset('siminfo_totsteps', data=self.steps)
+
+        # fuel and blanket density
+        dens_dict = {}
+        for key, value in self.mat_def_dict.items():
+            if float(value.split()[2]) < 0:
+                cat = 'mass'
+                dens_dict[key] = -1.0 * float(value.split()[2])
+            else:
+                cat = 'atomic'
+                dens_dict[key] = float(value.split()[2])
+        for key, value in dens_dict.items():
+            if key == self.driver_mat_name:
+                key = 'driver'
+            elif key == self.blanket_mat_name:
+                key = 'blanket'
+            self.f.create_dataset('siminfo_%s_%s_density' %(key, cat), data=value)
+
+        init_comp = self.read_dep(boc=True)
+        for key, value in init_comp.items():
+            if key == self.driver_mat_name:
+                key = 'driver'
+            elif key == self.blanket_mat_name:
+                key = 'blanket'
+            self.f.create_dataset('siminfo_%s_init_comp' %key, data=value)
```

Comment:
writing to the data set really should n't be in the saltproc class. There should be a separate database handling class.